x-pack/plugins/security_solution/public/common/components/autocomplete/helpers.ts

```diff
@@ -30,29 +30,33 @@ export const getOperators = (field: IFieldType | undefined): OperatorOption[] =>
   }
 };
 
-export function validateParams(params: string | undefined, type: string) {
+export const validateParams = (
+  params: string | undefined,
+  field: IFieldType | undefined
+): boolean => {
   // Box would show error state if empty otherwise
   if (params == null || params === '') {
     return true;
   }
 
-  switch (type) {
-    case 'date':
-      const moment = dateMath.parse(params);
-      return Boolean(moment && moment.isValid());
-    case 'ip':
-      try {
-        return Boolean(new Ipv4Address(params));
-      } catch (e) {
-        return false;
-      }
-    case 'number':
-      const val = parseFloat(params);
-      return typeof val === 'number' && !isNaN(val);
-    default:
-      return true;
-  }
-}
+  const types = field != null && field.esTypes != null ? field.esTypes : [];
+
+  return types.reduce<boolean>((acc, type) => {
+    switch (type) {
+      case 'date':
+        const moment = dateMath.parse(params);
+        return Boolean(moment && moment.isValid());
+      case 'ip':
+        try {
+          return Boolean(new Ipv4Address(params));
```

Comment:
This new object seems weird to me...Does this mean we only support ipv4 and not ipv6?

If this is only validating ipv4 but allowing ipv6 I am fine with it. If we begin hindering users or disallowing ipv6 though we are going to start to have issue relatively quickly.

Reply:
I did a bit of research and there isn't really great consensus on a good regex for ipv6. I didn't find a similar util like the one above for ipv4 in our code so I removed the validation here. I don't want to block users from adding exceptions. This can be something we can circle back on.